### DISCUSSION TITLE
Inquiry Regarding Support Status of InAppBrowser Plugin in Cordova

### DISCUSSION THREAD

**Discussion by dt237131** [apache/cordova#479] [General]
Hey Cordova community,

I hope you're doing well. 
I wanted to ask about the status of the InAppBrowser plugin. With Ionic declaring end of life for it, I'm curious if the Cordova community plans to continue supporting it or if there are alternative solutions recommended.

Any insights you could provide would be really helpful!

Thanks

**Comment by breautek** [comment]
Migrating this to Cordova Discussions.

**Comment by breautek** [comment]
> With Ionic declaring end of life for it, I'm curious if the Cordova community plans to continue supporting it or if there are alternative solutions recommended.

Apache Cordova has nothing to do with Ionic. Cordova itself is standalone and at this time there is no formal discussion on sunsetting or deprecating the plugin.

I'll say though depending on your use case, or how you use the in-app-browser, using Chrome Tabs / Safari View Controller (a third-party plugin is [available](https://www.npmjs.com/package/cordova-plugin-safariviewcontroller), but I don't know how to integrate with Ionic) might be a better fit.

Most users who displays web content, doesn't inject or want to communicate with the web content, and wants a more full-featured "browser" window would likely benefit from this. It's far closer to embedding a real browser into your app, but compared to a webview which the `in-app-browser` implements, it lacks communication features for security/privacy reasons.

If your app depends on injecting content into the webview, or want to be able to send messages from and to the in-app-browser webview, then you'll need to continue to use a webview solution like the in-app-browser plugin. If Ionic is deprecating their wrapper for it, that would mean figuring out how to maintain a wrapper like ionic used to do.

**Comment by jcesarmobile** [comment]
What Ionic is deprecating are several Cordova enterprise (paid) plugins.

That doesn’t affect Apache Cordova plugins, nor the wrappers for them. The old Ionic wrappers (`@ionic-native`) were renamed to `@awesome-cordova-plugins`  years ago as they were donated to the community. They are still being maintained by the community.

The enterprise plugins deprecation only affects a small number of Ionic paying customers and they have been advised by Ionic Customer Support team about paths for migration. Doesn’t affect any non paying Ionic users or non Ionic users.
